src/module/system/text-editor.ts

```diff
@@ -320,6 +320,62 @@ class TextEditorPF2e extends TextEditor {
         // Deduplicate traits
         const allTraits = Array.from(new Set(traits));
 
+        const types = params.type.split(",");
+        let adjustments = params.adjustment?.split(",") ?? ["0"];
+
+        if (types.length !== adjustments.length && adjustments.length > 1) {
+            ui.notifications.warn(game.i18n.localize("PF2E.InlineCheck.Errors.AdjustmentLengthMismatch"));
+            return null;
+        } else if (types.length > adjustments.length) {
+            adjustments = new Array(types.length).fill(adjustments[0]);
+        }
+
+        if (adjustments.some((adj) => adj !== "" && isNaN(parseInt(adj)))) {
+            ui.notifications.warn(game.i18n.localize("PF2E.InlineCheck.Errors.NonIntegerAdjustment"));
+            return null;
+        }
+
+        const buttons = types.map((type, i) =>
+            this.#createSingleCheck({
+                allTraits,
+                actor,
+                item,
+                inlineLabel,
+                params: { ...params, ...{ type, adjustment: adjustments[i] || "0" } },
+            })
+        );
+        if (buttons.length === 1) {
+            return buttons[0];
+        } else {
+            const checkGroup = document.createElement("div");
+            checkGroup.setAttribute("data-pf2-checkgroup", "");
+            for (const button of buttons) {
+                if (button === null) {
+                    // Warning should have been displayed already by #createSingleCheck
+                    return null;
```

Comment:
Does it make the typecheck tricky if you move all this to a 

`else if (buttons.some((b) => !b) { 
    return null; 
}``` 

case earlier on? If it does don't worry about it, if it doesn't, it might make it easier to read for the next person if you do that.

Reply:
Yeah, it currently serves as a type guard. I could use the suggestion and add a non-null assertion, but personally I like to avoid those wherever possible to get the most out of the type system.

Reply:
I suspect remeda might have a way to avoid such an assertion, but I won't force it for this.